File: src/main/scala/us/illyohs/sparkyforge/command/ICommand.java
package us.illyohs.sparkyforge.command;



import org.kitteh.irc.client.library.element.Channel;
import org.kitteh.irc.client.library.element.User;

public interface ICommand
{
    String name();

    String help();

    void execute(User user, Channel channel, String message, String[] args);

}

File: src/main/scala/us/illyohs/sparkyforge/command/ICommand.scala
package us.illyohs.sparkyforge.command

import org.kitteh.irc.client.library.element.Channel
import org.kitteh.irc.client.library.element.User

trait ICommand {

  def name: String

  def help: String

  def execute(user: User, channel: Channel, message: String, args: Array[String])
}